### PR TITLE
APB-7796 on submit

### PR DESCRIPTION
**Pull request by raw1697**
APB-7796 & APB-7979 completion

### CODE REVIEW

File: app/uk/gov/hmrc/agentservicesaccount/models/AmlsRequest.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import play.api.libs.json.Json
+
+import java.time.LocalDate
+
+case class AmlsRequest(ukRecord: Boolean,
+                                          supervisoryBody: String,
+                                          membershipNumber: String,
+                                          membershipExpiresOn: Option[LocalDate]
+                      )
```
Comment (r-melvin) [inline app/uk/gov/hmrc/agentservicesaccount/models/AmlsRequest.scala:27]: Can you fix this formatting please?

File: app/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -70,10 +72,36 @@ class CheckYourAnswersController @Inject()(amlsLoader: AMLSLoader,
...
   private[amls] def buildSummaryListItems(isUkAgent: Boolean, journey: UpdateAmlsJourney, lang: Locale): Seq[SummaryListData] = {
     for {
       body <- journey.newAmlsBody.map {
-        case body if isUkAgent => supervisoryBodies(body)
+        case body if isUkAgent =>
+          supervisoryBodies.collectFirst {
+            case (_, value) if value == body => value
+          }.getOrElse(body)
+
         case body => body
```
Comment (gitwojciech) [inline app/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersController.scala:105]: journey.newAmlsBody.map always returns newAmlsBody

File: app/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -47,20 +48,19 @@ import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import java.time.format.DateTimeFormatter
 import java.util.Locale
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 
 @Singleton
 class CheckYourAnswersController @Inject()(amlsLoader: AMLSLoader,
```
Comment (r-melvin) [inline app/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersController.scala:55]: amlsLoader no longer needs to be injected here

File: test/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersControllerSpec.scala [hunk 4]
```diff
@@ -113,7 +113,7 @@ class CheckYourAnswersControllerSpec extends PlaySpec with IdiomaticMockito with
     mockAmlsLoader.load(*[String]) returns Map("ABC" -> "Alphabet")
```
Comment (r-melvin) [inline test/uk/gov/hmrc/agentservicesaccount/controllers/amls/CheckYourAnswersControllerSpec.scala:113]: You don't need this mock either

**Review by r-melvin — APPROVED** [review]

**Pull request merged**